### ISSUE TITLE
Consider to rewrite shell scripts in Python

### ISSUE THREAD

**Issue by PragmaTwice** [apache/incubator-fury#1299]
TBH these scripts are hard to maintain (especially considering the difference between bash and zsh), we need a more structure language like python to make them readable and maintanable.

**Comment by LiangliangSui** [comment]
I totally agree with this, it can mask the differences in script syntax (e.g. windows script syntax is very unfriendly), and it is easier to write and code readable.

**Comment by chaokunyang** [comment]
Sounds good to me, but rewriting those scripts into python is not an easy work, and most scripts are used by CI only. I suggest we start from rewriting the scripts used by contributors everyday.

**Comment by chaokunyang** [comment]
This remind me of `PyTorch`. When you filing an issue, you can execute a python script from `PyTorch`, which can collect system information as much as it can, and print it. Then you can paste such information into the issue

**Comment by LiangliangSui** [comment]
So can we start by rewriting the C++ CI Build script in Windows? such as `run_ci.py`

**Comment by chaokunyang** [comment]
Yep, we can start from rewriting the C++ CI Build script in Windows
